elastic/kibana
# [ES Query] Save ECS keyword group by fields in AAD document

Related to https://github.com/elastic/kibana/issues/183220

## Summary

This PR saves ECS keyword group by fields in AAD document for ES query rule.

|Rule|Before|After|
|---|---|---|
|![image](https://github.com/user-attachments/assets/0ba8699d-44bf-4930-b3f0-33b4ed8a7e0c)|![image](https://github.com/user-attachments/assets/a98d70e5-213e-4dc3-93a7-fa509828d15b)|![image](https://github.com/user-attachments/assets/00a57945-1590-4286-8132-072e19d5866f)|
|![image](https://github.com/user-attachments/assets/8fe39de4-97ac-4a66-8c69-5c2b552b8264)|![image](https://github.com/user-attachments/assets/9f3ec653-fa11-45cc-add0-d1923e68547e)|![image](https://github.com/user-attachments/assets/41ce2028-54f0-4ca4-8d63-0d371819865f)|

### How to test
- Create some data with ECS fields
    - For example, you can use synthtrace command: `node scripts/synthtrace simple_trace.ts --local --live`
- Create an ES Query rule grouped by ECS and non-ECS fields
- In the generated alert, you should be able to see the ECS group by field but not the no-ECS ones
